entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/mapper/ConvertOperationBuilderImpl.java

```diff
@@ -54,7 +54,11 @@ public T convert() {
 
     @Override
     public ConvertOperationBuilder<T> excludeAttribute(String attributePath) {
-        if (!key.getSourceType().getRecursiveAttributes().containsKey(attributePath) && !key.getSourceType().getRecursiveSubviewAttributes().containsKey(attributePath)) {
+        if (
+                (key.getSourceType() != null)
+                && !key.getSourceType().getRecursiveAttributes().containsKey(attributePath)
+                && !key.getSourceType().getRecursiveSubviewAttributes().containsKey(attributePath)
+        ) {
             throw new IllegalArgumentException("Attribute '" + attributePath + "' could not be found on type: " + key.getSourceType().getJavaType().getName());
```

Comment:
I think it's more appropriate to use the target type here.
```suggestion
        if (!key.getTargetType().getRecursiveAttributes().containsKey(attributePath) && !key.getTargetType().getRecursiveSubviewAttributes().containsKey(attributePath)) {
                    throw new IllegalArgumentException("Attribute '" + attributePath + "' could not be found on type: " + key.getTargetType().getJavaType().getName());
```

and also in the `excludeAttributes(String...)` below. Previously it didn't make much difference because views had to be compatible i.e. both contain the same/similar attributes, but it's more correct to base this on the target attributes.